src/main/java/org/mule/tools/apikit/model/MuleDomain.java

```diff
@@ -44,6 +53,38 @@ public static MuleDomain fromInputStream(InputStream content) throws Exception {
     return new MuleDomain(content, httpListenerConfigs);
   }
 
+  public static MuleDomain fromConfigFile(File configFile) throws Exception {
+    try (InputStream configFileIS = new FileInputStream(configFile)) {
+      // This is necessary to prevent leaking file handles
+      InputStream content = cloneInputStream(configFileIS);
+      Document contentAsDocument = new SAXBuilder().build(content);
+      List<HttpListenerConfig> httpListenerConfigs = new HttpListenerConfigParser().parse(contentAsDocument);
+      return new MuleDomain(content, httpListenerConfigs);
+    }
+  }
+
+  public static MuleDomain fromDeployableArtifact(File artifact) throws Exception {
+    try (URLClassLoader cl = new URLClassLoader(new URL[] {artifact.toURI().toURL()}, MuleDomain.class.getClassLoader())) {
```

Comment:
why can't this be part of studio?

Reply:
I've discussed this with Pablo Cabrera at https://github.com/mulesoft/Mule-Tooling/issues/1166, he thinks this should be in the scaffolder as this is also a form of domains it should understand.

Reply:
why do we need an URLClassloader?